Makefile

```diff
@@ -0,0 +1,123 @@
+
+#Check for python3 install and virtual environment
+$(PYTHON3):
+	@if [ -z python3 ]; then \
+		$(FAIL);\
+		echo "Python 3 could not be found.";\
+		$(RESET_TERM);\
+		exit 2; \
+	fi
+	python3 -m venv $(VENV)
+
+
+#Install the project via setup.py
+.PHONEY: venv-install
+venv-install: $(CARAMEL_TOOL)
+$(CARAMEL_TOOL): $(PYTHON3) setup.py
+	$(VENV)/bin/pip3 install -e .
+
+
+#Basic tests that caramel can be CARAMEL_TOOL and run with test data, using environment variables for config
+.PHONEY: systest
+systest: export CARAMEL_INI = development.ini
+systest: export CARAMEL_DBURL = sqlite:///caramel.sqlite
+systest: export CARAMEL_LOG_LEVEL = 0
+systest: export CARAMEL_CA_CERT = $(VENV)/caramel.ca.cert
+systest: export CARAMEL_CA_KEY = $(VENV)/caramel.ca.key
+systest: export CARAMEL_CA = 127.0.0.1:6543
+systest: $(CARAMEL_TOOL)
+	@$(BOLD); echo "Removing $(CARAMEL_CA_CERT), $(CARAMEL_CA_KEY) and $(CARAMEL_DBURL)"; $(LINE); $(RESET_TERM)
+	rm -f $(CARAMEL_CA_CERT) $(CARAMEL_CA_KEY) $(CARAMEL_DBURL)
+	@$(BLR)
+
+	@$(BOLD); echo "Generate new CA cert and key with ca_test_input.txt"; $(LINE); $(RESET_TERM)
+	$(VENV)/bin/caramel_ca < ca_test_input.txt
+	@$(BLR)
+
+	@$(BOLD); echo "Create a new DB at $(CARAMEL_DBURL)"; $(LINE); $(RESET_TERM);
+	$(VENV)/bin/caramel_initialize_db
+	@$(BLR)
+
+	@timeout 2s $(VENV)/bin/pserve caramel_launcher.ini \
+		ca_cert=$$CARAMEL_CA_CERT \
+		ca_key=$$CARAMEL_CA_KEY \
+		http_port=6543 \
+		http_host=127.0.0.1 \
+		life_short=48 \
+		life_long=720 \
+		dburl=$$CARAMEL_DBURL \
+		log_level=INFO;\
+		if [ $$? -eq 124 ]; then \
+			$(PASS); $(LINE); echo "Caramel server started and timed out successfully";\
+		else \
+			$(FAIL); $(LINE); echo "System test failed: Caramel server exited before time out with exit code: $$?";\
+			exit 1;\
+		fi;
+	@$(BLR)
+
+	@timeout 2s $(VENV)/bin/caramel_autosign;\
+		if [ $$? -eq 124 ]; then \
+			$(PASS); $(LINE); echo "Autosign server started and timed out successfully";\
+		else \
+			$(FAIL); $(LINE); echo "System test failed: Autosign server exited before time out with exit code: $$?";\
+			exit 1;\
+		fi;
+	@$(BLR)
+
+	@$(BOLD);echo "Start new caramel server and caramel_autosign in the background";$(BLR)
+	setsid $(VENV)/bin/pserve caramel_launcher.ini \
+		ca_cert=$$CARAMEL_CA_CERT \
+		ca_key=$$CARAMEL_CA_KEY \
+		http_port=6543 \
+		http_host=127.0.0.1 \
+		life_short=48 \
+		life_long=720 \
+		dburl=$$CARAMEL_DBURL \
+		log_level=INFO\
+		>/dev/null 2>&1 < /dev/null & \
+	echo $$! > $(VENV)/server.pid
+
+	setsid $(VENV)/bin/caramel_autosign >/dev/null 2>&1 < /dev/null &\
+	echo $$! > $(VENV)/autosign.pid
+	sleep 2s #Give server some time to start
+	@$(BLR)
+
+	@ $(BOLD); echo "Use client-example.sh to first upload our CSR, wait for it \
+	to get processed,then call it again to confirm the server stored our CSR"; $(BLR)
+	chmod +x client-example.sh
+	./client-example.sh $(VENV)
+	sleep 2s
+	./client-example.sh $(VENV)
+	kill `cat $(VENV)/server.pid` `cat $(VENV)/autosign.pid`
```

Comment:
Mixing shell syntax and make syntax here. $(shell   )  is usually recommended for shell syntax. vs backticks.


Also. Since you have PID files after launching, you can make the launch server a make target that creates the PID file, which in turn willl clean out the dep chain here.